generator-api/src/main/scala/com/lightbend/lagom/spec/ResourceUtils.scala

```diff
@@ -42,4 +43,25 @@ object ResourceUtils {
     }
   }
 
+  /**
+   * @param folder an absolute folder
+   * @param filename a relative path to a file (may contain File.separators)
+   * @param fileContents
+   * @return
+   */
+  def writeFile(folder: File, filename: String, fileContents: String): File = {
+    val path = Paths.get(folder.getAbsolutePath, filename)
+    val theFolder = path.toFile.getParentFile
+    while (!theFolder.exists()) {
```

Comment:
It seems like this could get stuck in an infinite loop if there's a problem that can't be resolved by retrying.